.github/workflows/tests.yml

```diff
@@ -2,27 +2,49 @@ name: Tests
 
 on: [push, pull_request]
 
+
 jobs:
   Tests:
 
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v1
-    - name: Install dependencies
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v1
+      id: cache
+      with:
+        path: /usr/share/miniconda/envs/improver
+        key: ${{ runner.os }}-conda-improver-${{ hashFiles('**/environment.yml') }}
+        restore-keys: |
+          ${{ runner.OS }}-conda-improver-
+
+    - name: Resolve and install dependencies
+      if: steps.cache.outputs.cache-hit != 'true'
       run: |
-        $CONDA/bin/conda env update -q --file environment.yml --name base
-        $CONDA/bin/pip install clize
-        $CONDA/bin/conda list --export
+        source '/usr/share/miniconda/etc/profile.d/conda.sh'
```

Comment:
why is this needed, given that conda is amongst [installed packages on github runners](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners)?

Reply:
A known problem
https://github.com/conda/conda/issues/7980

Reply:
writing to .bashrc or similar might be a neater way around this repetition

Reply:
Couldn't we just set the `PATH` in the job? That's what `conda activate` boils down to, I think.

https://help.github.com/en/actions/automating-your-workflow-with-github-actions/development-tools-for-github-actions#add-a-system-path-add-path